src/Filter.ts

```diff
@@ -0,0 +1,110 @@
+export interface FilterArgs {
+	[ name: string ]: Filter | RegExp | string | (() => boolean);
+}
+
+export default class Filter {
+	[index: string]: any;
+	type: string;
+	args: any[];
+
+	constructor();
+	constructor(filterArg: string);
+	constructor(filterArg: () => any);
+	constructor(filterArg: FilterArgs);
+	constructor(filterArg?: any) {
```

Comment:
I combined the constructor declarations using the union type, and realized that the any type wasn't actually needed.